aws/resource_aws_instance_test.go

```diff
@@ -1060,6 +1060,98 @@ func TestAccAWSInstance_volumeTagsComputed(t *testing.T) {
 	})
 }
 
+// This test attempts to expose the issues identified in #5080, #5609, #5878.
+//
+// All issues are associated with the use of aws_ebs_volumes and aws_volume_attachment
+// in managing additional EBS volumes (other than the root volume) attached to a given
+// instance.
+//
+// There are basically two (2) types of failure modes:
+//   1. ebs_block_device is not in configuration but ends up set and often presisted
+//      to state as the result of AWS interrogation.
+//   2. Instance volume_tags and attached volume tags settings drift between
+//      the tag sets, sometimes in a benign manner, and other times altering
+//      data.
+//
+// The error condition encountered can vacillate from one to the other between
+// instantiations of a given configuration or changes to a given configuration
+// due to what appears to be some sort of race condition that affects the ordering
+// of data returned from AWS.  As such it exposes flaws with the way the affected
+// data is handled.
+//
+// Running this test case (without the fix that will accompany this test) will
+// demonstrate the vacilation in error conditions between executions.
+//
+func TestAccAWSInstance_ebsVolumeTags(t *testing.T) {
+	var v ec2.Instance
+	var x ec2.Volume
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckInstanceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckInstanceConfigWithAttachedEbs,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckInstanceExists("aws_instance.foo", &v),
```

Comment:
We are trying to have all resources in tests named `test`, `test2`, etc rather than `foo`, `bar`. This is a new convention for us but that consistency will really help us going forward - could you change this throughout these new test configs you've written? Thanks!

```suggestion
					testAccCheckInstanceExists("aws_instance.test", &v),
```